nym-vpn-desktop/src-tauri/src/commands/db.rs

```diff
@@ -0,0 +1,138 @@
+use serde::{Deserialize, Serialize};
+use tauri::State;
+use tracing::{debug, instrument, trace};
+use ts_rs::TS;
+
+use crate::{
+    db::{Db, DbError, JsonValue, Key},
+    error::{CmdError, CmdErrorSource},
+    states::app::{NodeLocation, VpnMode},
+};
+
+#[derive(Default, Serialize, Deserialize, Debug, Clone, TS)]
+#[ts(export)]
+pub enum UiTheme {
+    Dark,
+    #[default]
+    Light,
+    System,
+}
+
+#[derive(Default, Serialize, Deserialize, Debug, Clone, TS)]
+#[ts(export)]
+pub struct AppData {
+    pub monitoring: Option<bool>,
+    pub autoconnect: Option<bool>,
+    pub entry_location_enabled: Option<bool>,
+    pub ui_theme: Option<UiTheme>,
+    pub ui_root_font_size: Option<u32>,
+    pub vpn_mode: Option<VpnMode>,
+    pub entry_node_location: Option<NodeLocation>,
+    pub exit_node_location: Option<NodeLocation>,
+}
+
+#[instrument(skip_all)]
+#[tauri::command]
+pub async fn db_get_batch(db: State<'_, Db>) -> Result<AppData, CmdError> {
```

Comment:
I feel like this function would have been way simpler if instead we just stored `AppData` as a single serialized struct in say a `.toml` file. or even a stupid `.json`. I said it before, I feel like `sled` is an overkill. we don't need its performance here.

plus I reckon it will be way more user-readable

Reply:
The whole purpose of this refactoring is to allow crud ops on key/value pairs. To have the flexibility needed by UI interactions.
This is why we don't want to store the entire app data as a single block of data. As we want to be able to make single value update/read and so on.

> we don't need its (sled) performance here

So I agree, and I didn't say it was the main reason to switch on sled. But free performance isn't something always positive and good to take?

> plus I reckon it will be way more user-readable

Yes but we don't need that, only code access is needed. Not "human" access. It's app internal data; not like a config file, where dev/user would have direct interaction with.